editor/find_in_files.cpp

```diff
@@ -721,7 +723,9 @@ void FindInFilesPanel::_on_result_found(String fpath, int line_number, int begin
 
 	// Trim result item line
 	int old_text_size = text.size();
-	text = text.strip_edges(true, false);
+	if (_finder->is_whole_words()) {
```

Comment:
Why strip text here? `text` is supposed to be the entire line where the result was found, not the search pattern

Reply:
I am pretty busy at work ATM so I will have to get back to this once things calm down. But to answer your question, when testing this I was getting weird results where extra white space was being selected on the line if that conditional is not there.

I'll run some tests when I get the chance to replicate it and show it here. The conditional call to strip lines ensures everything looks "pretty" and uniform between the two options.

Reply:
I would try to figure out why, because adding this condition doesn't seem to make sense. It should not be required.
The trim actually wasn't present when I initially implemented this functionality, it was added in dc7f5557839b27831d284d44525a4b4815f4a46e, not sure why.

Maybe you are getting this issue because the size of the match highlighting is determined *not* from the actual text being searched, but some label instead:
```
	match_rect.size.x = font->get_string_size(_search_text_label->get_text()).x;
```
And that label is set *before* `FindInFiles::start()` is called, see `FindInFilesPanel::start_search()`. Which means it won't have the trimming you added in the `FindInFiles` class.
Using a label meant for display here is a design mistake and should use the real pattern instead, or use the actual width of the search result text (this was also introduced by dc7f5557839b27831d284d44525a4b4815f4a46e)